py3status/modules/vpn_status.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+"""
+Drop-in replacement for i3status run_watch VPN module.
+
+Expands on the i3status module by displaying the name of the connected vpn
+using the NetworkManager python module.
+
+Configuration parameters:
+    cache_timeout: How often we refresh this module in seconds.
+        (default 10)
+    check_pid: If true, act just like the default i3status module.
+        (default False)
+    format: Format of the output.
+        (default 'VPN: {name}')
+    pidfile: Same as i3status.conf pidfile, checked when check_pid is True.
+        (default '/sys/class/net/vpn0/dev_id')
+
+Format string parameters:
+    {name} The name and/or status of the VPN.
+
+Requires:
+    python-NetworkManager
+
+@author Nathan Smith <nathan AT praisetopia.org>
+"""
+
+try:
+    import NetworkManager
```

Comment:
No, it's not really needed, mostly just easier on the eyes. Here, I wrote:
```
for conn in NetworkManager.NetworkManager.ActiveConnections:
    # do stuff
```
The same code with pure dbus is something like:

```
bus = dbus.SystemBus()
m_proxy = bus.get_object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager")
manager = dbus.Interface(m_proxy, "org.freedesktop.NetworkManager")
mgr_props = dbus.Interface(m_proxy, "org.freedesktop.DBus.Properties")

s_proxy = bus.get_object("org.freedesktop.NetworkManager", "/org/freedesktop/NetworkManager/Settings")
settings = dbus.Interface(s_proxy, "org.freedesktop.NetworkManager.Settings")

active = mgr_props.Get("org.freedesktop.NetworkManager", "ActiveConnections")
for a in active:
    a_proxy = bus.get_object("org.freedesktop.NetworkManager", a)
    a_props = dbus.Interface(a_proxy, "org.freedesktop.DBus.Properties")
    # do stuff
```